test/provisioner/provision.go

```diff
@@ -64,6 +74,98 @@ type InstallOverlay interface {
 	Edit(ctx context.Context, cfg *envconf.Config, properties map[string]string) error
 }
 
+func runCommand(command string, stdout io.Writer, stderr io.Writer, args ...string) error {
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	fmt.Printf("Running command: %s %v\n", command, args)
+
+	if err := cmd.Run(); err != nil {
+		err = fmt.Errorf(fmt.Sprintf("Error running command: %s %v - %s", command, args, err))
+
+		log.Errorf("%v", err)
+		return err
+	}
+
+	return nil
+}
+
+func NewKeyBrokerService(clusterName string) (*KeyBrokerService, error) {
+	// Clone kbs repo
+	repoURL := "https://github.com/confidential-containers/kbs"
+
+	if err := runCommand("git", os.Stdout, os.Stderr, "clone", repoURL); err != nil {
+		return nil, err
+	}
+
+	log.Info("creating key.bin")
+
+	// Create secret
+	content := []byte("This is my cluster name: " + clusterName)
+	filePath := "kbs/kbs/config/kubernetes/overlays/key.bin"
+	// Create the file.
+	file, err := os.Create(filePath)
+	if err != nil {
+		err = fmt.Errorf("Error creating file: %w\n", err)
+		log.Errorf("%v", err)
+		return nil, err
+	}
+	defer file.Close()
+
+	// Write the content to the file.
+	_, err = file.Write(content)
+	if err != nil {
+		err = fmt.Errorf("Error writing to the file: %w\n", err)
+		log.Errorf("%v", err)
+		return nil, err
+	}
+
+	k8sCnfDir, err := os.Getwd()
+	if err != nil {
+		err = fmt.Errorf("Error getting the current working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return nil, err
+	}
+	fmt.Println(k8sCnfDir)
+
+	keyFile := filepath.Join(k8sCnfDir, "kbs/kbs/config/kubernetes/overlays/key.bin")
+	if _, err := os.Stat(keyFile); os.IsNotExist(err) {
+		err = fmt.Errorf("key.bin file does not exist")
+		log.Errorf("%v", err)
+		return nil, err
+	}
+
+	kbsCert := filepath.Join(k8sCnfDir, "kbs/kbs/config/kubernetes/base/kbs.pem")
+	if _, err := os.Stat(kbsCert); os.IsNotExist(err) {
```

Comment:
If the file exists we should probably not overwrite it and bail out

Reply:
i think its fine as it gets updated with latest cluster name